lms/static/scripts/frontend_apps/components/FilePickerApp.tsx

```diff
@@ -69,6 +70,35 @@ function contentDescription(content: Content) {
   }
 }
 
+/**
+ * Fetch additional configuration needed by the file picker app.
+ *
+ * This is needed when transitioning to the file picker from another route.
+ *
+ * Returns the result of merging {@link config} with the configuration for
+ * the file picker app.
+ */
+export async function loadFilePickerConfig(
+  config: ConfigObject
+): Promise<ConfigObject> {
+  if (!config.editing) {
+    throw new Error('Assignment editing config missing');
+  }
```

Comment:
This will only happen if a mistake was made by us and the frontend tries to do a route transition which is not possible given the configuration it launched with. In other words, this check is effectively asserting something which _should_ be true.